### ISSUE TITLE
ERROR: "FILES NOT RENAMED!"

### ISSUE THREAD

**Issue by Au2mater** [OlegLustenko/obsidian-bulk-rename#37]
I am trying to replace hyphens _ with dashes, but I keep on getting the message "FILES NOT RENAMED!" and "WARNING: YOU MUST CREATE FOLDER BEFORE MOVING INTO IT" I am not sure what this means and how to fix it.
![image](https://github.com/OlegLustenko/obsidian-bulk-rename/assets/93315669/1dc41523-3a52-4fd7-a2d6-d6eb720f1efc)
![image](https://github.com/OlegLustenko/obsidian-bulk-rename/assets/93315669/c274d8ce-bf97-421d-8308-92faf6f3d91d)
Here's an exmaple of a folder where there error above occurs:

![image](https://github.com/OlegLustenko/obsidian-bulk-rename/assets/93315669/c06253cf-2ed7-44a7-a88a-53d2b09a943d)
![image](https://github.com/OlegLustenko/obsidian-bulk-rename/assets/93315669/bb7c60c5-6bdc-470d-b2bf-69d5c41e8ba3)

The error seems to occur when the rename affects both folder and note name.
I suggest the code be updated to first rename the folder structure then update the update file names.

**Comment by NOOBLINUXUSER** [comment]
I am having the same issue But idek how to create that new folder they are asking me to do.

**Comment by NOOBLINUXUSER** [comment]
> I am trying to replace hyphens _ with dashes, but I keep on getting the message "FILES NOT RENAMED!" and "WARNING: YOU MUST CREATE FOLDER BEFORE MOVING INTO IT" I am not sure what this means and how to fix it. ![image](https://private-user-images.githubusercontent.com/93315669/322409733-1dc41523-3a52-4fd7-a2d6-d6eb720f1efc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIyODc3MjIsIm5iZiI6MTczMjI4NzQyMiwicGF0aCI6Ii85MzMxNTY2OS8zMjI0MDk3MzMtMWRjNDE1MjMtM2E1Mi00ZmQ3LWEyZDYtZDZlYjcyMGYxZWZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTIyVDE0NTcwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdjMzdjZjBjNDU2Mzc4YmUzZGY5ZGJjNmQ0YzFjMDk0ZDkyNmRhMDJkMzhiZmE3NjBkNzAxODNiZDI1MDFlZjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.gIFySAI3NrQK9TiUJR11J2GbB4LwUtJm3Ueenjqj4X4) ![image](https://private-user-images.githubusercontent.com/93315669/322409748-c274d8ce-bf97-421d-8308-92faf6f3d91d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIyODc3MjIsIm5iZiI6MTczMjI4NzQyMiwicGF0aCI6Ii85MzMxNTY2OS8zMjI0MDk3NDgtYzI3NGQ4Y2UtYmY5Ny00MjFkLTgzMDgtOTJmYWY2ZjNkOTFkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTIyVDE0NTcwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI1YTdiOWQ0OTI0NGMwODA2YmNhNzJjZmJjMDk3ZGFhNmJjZTRlOTk5ZDcyY2I1NDdmNzFkNDMzYmRhMTU3NDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.QSUP2mqQbqL7QA6o3J7WklNrXEja2kgvQ-xqO3WMCWg) Here's an exmaple of a folder where there error above occurs:
> 
> ![image](https://private-user-images.githubusercontent.com/93315669/322411580-c06253cf-2ed7-44a7-a88a-53d2b09a943d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIyODc3MjIsIm5iZiI6MTczMjI4NzQyMiwicGF0aCI6Ii85MzMxNTY2OS8zMjI0MTE1ODAtYzA2MjUzY2YtMmVkNy00NGE3LWE4OGEtNTNkMmIwOWE5NDNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTIyVDE0NTcwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdmNDhlMDZiZDQxMDgzMmM1MzQ1MzU4ZTJiNWE4MWQ3NzhkM2FiMTgxMmY1YWY3OGU4ZWU2OTExYWUwYzdiMzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.yjmAFkjQMEDYLAnUPkqk8JG5ZghLqagFtJYx9IgA8NQ) ![image](https://private-user-images.githubusercontent.com/93315669/322411824-bb7c60c5-6bdc-470d-b2bf-69d5c41e8ba3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIyODc3MjIsIm5iZiI6MTczMjI4NzQyMiwicGF0aCI6Ii85MzMxNTY2OS8zMjI0MTE4MjQtYmI3YzYwYzUtNmJkYy00NzBkLWIyYmYtNjlkNWM0MWU4YmEzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTIyVDE0NTcwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJhMzZjYTlmOWEyZThmOWNmOTQ2ZjcwZTA3OWFiYTI3ZWE2YTFjYzkwYzQ1N2YzNjA5Yzc5NTVjMjg0YWIwMzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3tbutbRdRV0eLaAHpu-zky27Zn7k6bLcWYNL9N7K5Lc)
> 
> The error seems to occur when the rename affects both folder and note name. I suggest the code be updated to first rename the folder structure then update the update file names.

were you able to fix this?

**Comment by Saashiguma** [comment]
I know this is kind of old, but I had the same problem. However, I was able to solve it, so I'm leaving this here for anybody who ended up here after searching.

My issue was how I attempted to rename a lot of files _including folders_ to match each other (Zettelkasten style), so I had to change "2F" in every file to be "2D" instead. The only issue is my folders with these new names didn't exist yet, and Bulk Rename didn't create them for me.

My fix was to manually create the directories I was trying to rename beforehand. I had to create a directory with the same name as the master one with 2D instead of 2F. After that, I had to create matching directories inside of this new directory (2D2, 2D8, and so on).

Only after this was I able to run Bulk Rename just fine. What tipped me off on this being the problem was how I tried creating a master directory with the 2D name at the beginning first with no directories inside of it, but only a couple of files successfully renamed into it.

In short, create folders and directories for the files you plan to rename beforehand, including any folders within the folders. Everything should work just fine after that.
